tests/post/steps/test_solutions.py

```diff
@@ -0,0 +1,383 @@
+    # and ensure the archive path is at the right location
+    with host.sudo():
+        assert name in host.file(SOLUTION_CONFIGURATION_FILE).content_string, \
+            'No {} Solution ISO path found in Solution Config {}'.format(
+                name, SOLUTION_CONFIGURATION_FILE
+            )
+
+    def _wait_for_imported_archive():
+
+        configmap = get_configmap(
+            k8s_client, SOLUTION_CONFIGMAP, SOLUTION_NAMESPACE
+        )
+        assert configmap is not None, \
+            'Failed to read Solution ConfigMap {}'.format(SOLUTION_CONFIGMAP)
+
+        data = configmap.get('data', {})
+        assert any(k == name for k, v in data.items()), \
+            '{} Configmap does not contain solution {}'.format(
+                SOLUTION_CONFIGMAP, name
+            )
+    utils.retry(
+        _wait_for_imported_archive,
+        times=10,
+        wait=5,
+        name="wait for imported archive"
+    )
+
+
+@then(parsers.parse(
+    "Solution '{name}' version '{solution_version}' is available"))
+def is_available_solution(host, request, name, solution_version):
+    available_solution = get_available_solution(host)
+    assert name in available_solution
+    for k, v in available_solution.items():
+        if k == name:
+            for items in v:
+                assert items.get('version', None) == solution_version
+
+
+@then(parsers.parse(
+    "Solution '{name}' version '{solution_version}' is activated"))
+def is_activated_solution(host, name, solution_version, k8s_client):
+    # Solution is marked activated in Solutions ConfigMap
+
+    def _wait_for_activated_solution():
+        configmap = get_configmap(
+            k8s_client, SOLUTION_CONFIGMAP, SOLUTION_NAMESPACE
+        )
+        assert name in configmap.get('data', {}), \
+            '{} Configmap does not contain {}'.format(SOLUTION_CONFIGMAP, name)
+
+        # Solution is marked activated in the Pillars
+        available_solutions = get_available_solution(host)
+        for k, v in available_solutions.items():
+            if k == name:
+                for items in v:
+                    assert bool(items.get('active', None))
+                    assert items.get('version', None) == solution_version
+    utils.retry(
+        _wait_for_activated_solution,
+        times=10,
+        wait=5,
+        name="wait for activated solution"
+    )
+
+
+@then(parsers.parse("CRD '{name}' exists in Kubernetes API"))
+def is_available_crds(request, host, name):
+    with host.sudo():
+        res = host.run((
+            'kubectl --kubeconfig=/etc/kubernetes/admin.conf '
+            'get crd {}'
+        ).format(name))
+        assert res.rc == 0, res.stdout
+
+
+@then(parsers.parse("solution environment '{name}' is available"))
+def read_solution_environment(k8s_client, name):
+    solution_environment_obj = get_environment(k8s_client, name)
+    assert any(name == env.metadata.name for env in solution_environment_obj)
```

Comment:
Environment name != namespace name, anyway it should be:
```suggestion
    assert get_environment(k8s_client, name) is not None
```